src/Symfony/Component/Message/Debug/LoggingMiddleware.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Message\Debug;
+
+use Symfony\Component\Message\MiddlewareInterface;
+use Psr\Log\LoggerInterface;
+
+/**
+ * @author Samuel Roze <samuel.roze@gmail.com>
+ */
+class LoggingMiddleware implements MiddlewareInterface
+{
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    public function __construct(LoggerInterface $logger)
+    {
+        $this->logger = $logger;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function handle($message, callable $next)
+    {
+        $this->logger->debug('Starting processing message', array(
```

Comment:
`... message "'.get_class($message).'"'`? (as for logs below).

Reply:
I don't think we should have that as part of the log message (as it's in the context) but happy to change if more people believe it should be the case.

Reply:
it would make identifying messages a bit easier (i.e. in the web profiler log panel, which would boldify the message class :))

Reply:
Only if the message is `Starting processing {class} message` and we add `class` as context I believe. Which... makes sense actually 👍